src/test/java/uk/gov/hmcts/reform/laubackend/idam/controllers/UserDeletionAuditControllerGetAllTest.java

```diff
@@ -0,0 +1,75 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import uk.gov.hmcts.reform.laubackend.idam.dto.DeletionLogAllUsersRequestParams;
+import uk.gov.hmcts.reform.laubackend.idam.insights.AppInsights;
+import uk.gov.hmcts.reform.laubackend.idam.response.UserDeletionGetResponse;
+import uk.gov.hmcts.reform.laubackend.idam.service.UserDeletionAuditService;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.laubackend.idam.insights.AppInsightsEvent.GET_ALL_DELETED_ACCOUNTS_INVALID_REQUEST_EXCEPTION;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+class UserDeletionAuditControllerGetAllTest {
+    private static final String USERID = "1234";
+    private static final String EMAIL = "john.smith@example.net";
+    private static final String START_TIMESTAMP = "2023-05-23T07:34:12";
+
+    @Mock
+    private UserDeletionAuditService userDeletionAuditService;
+
+    @Mock
+    private AppInsights appInsights;
+
+    @InjectMocks
+    private UserDeletionAuditController controller;
+
+    @Test
+    void shouldReturnResponseForGetAllUsersRequest() {
+        final UserDeletionGetResponse response = mock(UserDeletionGetResponse.class);
+        when(userDeletionAuditService.getAllDeletedUsers(any())).thenReturn(response);
+        DeletionLogAllUsersRequestParams params = new DeletionLogAllUsersRequestParams(
+            "10000",
+            "1",
+            ""
+        );
+        final ResponseEntity<UserDeletionGetResponse> responseEntity = controller
+            .getAllDeletedUsers(null, null, params);
+        verify(userDeletionAuditService, times(1)).getAllDeletedUsers(params);
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode(), "Http status code mismatch");
+    }
+
+    @Test
+    void shouldReturnBadRequestOnMissingRequiredArg() {
+        DeletionLogAllUsersRequestParams params = new DeletionLogAllUsersRequestParams(
+            null,
+            null,
+            null
+        );
+        var response = controller.getAllDeletedUsers(null, null, params);
+        assertResponseIsBadRequest(response, "Http status code mismatch");
+        verify(appInsights, times(1))
+            .trackEvent(eq(GET_ALL_DELETED_ACCOUNTS_INVALID_REQUEST_EXCEPTION.toString()), anyMap());
+    }
+
+    void assertResponseIsBadRequest(ResponseEntity<UserDeletionGetResponse> response, String message) {
+        HttpStatusCode statusCode = response.getStatusCode();
```

Comment:
Do you need to create a method with a single usage?

Reply:
method has been removed and used in code itself